src/invidious/routes/video_playback.cr

```diff
@@ -242,31 +242,25 @@ module Invidious::Routes::VideoPlayback
   # YouTube /videoplayback links expire after 6 hours,
   # so we have a mechanism here to redirect to the latest version
   def self.latest_version(env)
-    if env.params.query["download_widget"]?
-      download_widget = JSON.parse(env.params.query["download_widget"])
+    id = env.params.query["id"]?
+    itag = env.params.query["itag"]?.try &.to_i?
 
-      id = download_widget["id"].as_s
-      title = URI.decode_www_form(download_widget["title"].as_s)
-
-      if label = download_widget["label"]?
-        return env.redirect "/api/v1/captions/#{id}?label=#{label}&title=#{title}"
-      else
-        itag = download_widget["itag"].as_s.to_i
-        local = "true"
-      end
+    # Sanity checks
+    if id.nil? || id.size != 11 || !id.matches?(/^[\w-]+$/)
```

Comment:
```suggestion
    if id.nil? || !id.matches?(/^[\w-]{11}$/)
```
I'm not sure if that's the actual way to do it, but I'm the reviewer not the coder, I don't have to get it right 😄

Reply:
Yup, that's valid! However, checking the size first is done on purpose: if the string's length is not 11, we don't need to run the regex engine against the string. That saves a small amount of time.

I don't know if it has an important impact, though, given how high level Crystal is. It'd probably be faster to do a loop on the string's bytes and count & check at the same time.

(Yeah, my C habits may have a hard time, but I can't stop thinking that small optimizations like those could still make quite a difference at a larger scale).